### PR TITLE
Makes  use newly exposed endpoing on Gate to get base OS options.

### PR DESCRIPTION
**Pull request by ttomsu**
This PR needs to be merged after https://github.com/spinnaker/gate/pull/166 and https://github.com/spinnaker/orca/pull/687

Google/AWS base OS options:
base-os

Rosco disabled (Netflix-only) base OS options:
base-os-rosco-disabled

Error screenshot if specified base OS is not found:
base-os-error

@anotherchrisberry 
@spinnaker/reviewers

Tag: https://github.com/spinnaker/spinnaker/issues/597

### CODE REVIEW

File: app/scripts/modules/core/pipeline/config/stages/bake/bakery.service.js [hunk 1]
```diff
@@ -15,8 +16,16 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.bake.service', []
       }
     }
 
-    function getBaseOsOptions() {
-      return $q.when(['trusty', 'ubuntu', 'centos']);
+    function getBaseOsOptions(provider) {
+      if (provider) {
+        return getBaseOsOptions().then(function(accounts) {
+          return _.find(accounts, { cloudProvider: provider });
+        });
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/stages/bake/bakery.service.js]: `accounts` is probably not the right variable name, maybe `options`?

Comment (ttomsu) [inline app/scripts/modules/core/pipeline/config/stages/bake/bakery.service.js]: Renamed. Amended commit.

**Comment by anotherchrisberry** [comment]
Cool - LGTM - nice work!

**Comment by zanthrash** [comment]
@ttomsu LGTM - Can you rebase and I'll merge

**Comment by anotherchrisberry** [comment]
The other two PRs need to be merged and deployed before we merge this in.

**Comment by duftler** [comment]
Looks great.

**Pull request merged**